GO_SOPT_Seminar_Assignment/Local/TVING/Classes & Structure/MainViewContentsCell.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  ContentsCollectionViewCell.swift
+//  GO_SOPT_Seminar_Assignment
+//
+//  Created by 김다예 on 2023/04/29.
+//
+
+import UIKit
+
+class MainViewContentsCell: UICollectionViewCell {
+    
+    static let identifier = "ContentsCell"
+    
+    private let contentsStackView = UIStackView()
+    private let contentsImg = UIImageView()
+    private let contentsName = UILabel()
+    private let contentsSubName = UILabel()
+    
+    var pageIndicatorNum: Int = 0
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        setStyle()
+        setHierarchy()
+        setLayout()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+}
+
+extension MainViewContentsCell {
+    
+    func setStyle() {
+        contentsStackView.do {
+            $0.axis = .vertical
+            $0.distribution = .fill
+        }
+        contentsName.do {
+            $0.font = .tvingMedium(ofSize: 12)
+            $0.textColor = .tvingGray2
+        }
+        contentsSubName.do {
+            $0.font = .tvingRegular(ofSize: 10)
+            $0.textColor = .tvingGray2
+            $0.numberOfLines = 2
+        }
+    }
+    
+    func setHierarchy() {
+        contentView.addSubviews(contentsStackView)
+        contentsStackView.addArrangedSubviews(contentsImg,
+                                              contentsName,
+                                              contentsSubName)
+    }
+    
+    func setLayout() {
+        contentsStackView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        contentsImg.snp.makeConstraints {
+            $0.leading.trailing.equalToSuperview()
+        }
+        contentsName.snp.makeConstraints {
+            $0.height.equalTo(20)
+            $0.leading.trailing.equalToSuperview()
+        }
+        contentsSubName.snp.makeConstraints {
+            $0.height.equalTo(25)
+            $0.leading.trailing.equalToSuperview()
+        }
+    }
+    
+    func configureCell(_ contents: Contents) {
+        contentsImg.image = contents.image
+        contentsName.text = contents.name
+        contentsSubName.text = contents.subName
+        
+        if (contentsName.text == nil) {
+            contentsName.isHidden = true
+        } else {
+            contentsName.isHidden = false
+        }
+        if (contentsSubName.text == nil) {
+            contentsSubName.isHidden = true
+        } else {
+            contentsSubName.isHidden = false
+        }
+    }
```

Comment:
```suggestion
    func configureCell(_ contents: Contents) {
        contentsImg.image = contents.image
        
        if let name = contents.name {
            contentsName.text = name
        } else {
            contentsName.isHidden = true
        }
        
        if let subName = contents.name {
            contentsSubName.text = subName
        } else {
            contentsSubName.isHidden = true
        }
    }
```

위에 contentsName.text = contents.name 을 하고
그 아래에 또 contentsName.text == nil 을 확인한 뒤 값이 있으면 isHidden 을 false 로 한다~
는 코드의 중복이 조금 있는 것 같습니다..!!

제 나름대로 코드 양을 조금 줄이려고 노력해봤는데 잘 됐는지는 모르겠네욥!!